### PR TITLE
Basic integration with AI.JSX & FixieCorpus

### PR DESCRIPTION
**Pull request by farzadab**
This PR adds a minimal integration with AI.JSX and uses FixieCorpus with a fixed `corpusId` to load the existing Pitchfork Economics podcast corpus.

To run it you should set the following env variables:
```bash
export FIXIE_API_URL="https://mike-index-audio-pgaenaxiea-uc.a.run.app/api"
export FIXIE_API_KEY=.... # your regular (prod) key should work, but if not then you can get one from https://mike-index-audio-pgaenaxiea-uc.a.run.app/profile

yarn workspace webapp dev  # now start the app
```

Then you can go to a podcast page to see the SearchBox (e.g. http://localhost:3000/podcasts/random_podcast_name).

To see the preview, go to: https://podverse-git-farzad-fixie-corpus-call-fixie-ai.vercel.app/podcasts/pitchfork

### CODE REVIEW

File: packages/webapp/src/app/layout.tsx [hunk 1]
```diff
@@ -11,7 +12,9 @@ export const metadata = {
 export default function RootLayout({ children }: { children: React.ReactNode }) {
   return (
     <html lang="en">
-      <body className={inter.className}>{children}</body>
+      <body className={inter.className}>
+        <PageBase>{children}</PageBase>
```
Comment (farzadab) [inline packages/webapp/src/app/layout.tsx:16]: The `PageBase` (better name ideas?) allows for a uniform structure (e.g. header + theme) for all pages.

File: packages/webapp/src/app/page.tsx [hunk 2]
```diff
@@ -1,11 +1,5 @@
...
 import { MainApp } from '@/components/mainapp';
 
 export default function Home() {
-  return (
-    <main className="flex min-h-screen flex-col items-center justify-start p-24">
-      <AppHeader />
-      <MainApp />
-    </main>
-  );
+  return <MainApp />;
```
Comment (farzadab) [inline packages/webapp/src/app/page.tsx:4]: Moved this stuff to `page-base` component.

File: packages/webapp/src/app/podcasts/[podcast_id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { SearchBox } from '@/components/searchbox';
+
+export default function PodcastPage({ params }: { params: { podcast_id: string } }) {
+  return (
+    <div className="w-full max-w-5xl mt-8">
+      <div className="w-full flex justify-center mt-8">
+        <SearchBox podcast_id={params.podcast_id} endpoint={'../api/podcasts/query'} />
```
Comment (farzadab) [inline packages/webapp/src/app/podcasts/[podcast_id]/page.tsx]: Gotta provide a `corpusId` from KV here. Potentially we can have icons for the podcasts too.

File: packages/webapp/src/components/mainapp.tsx [hunk 4]
```diff
@@ -9,9 +6,6 @@ export function MainApp() {
       <div className="font-mono text-md p-4 border-2 border-slate-700 rounded-md">
         Podverse is an AI-powered podcast search engine.
       </div>
-      <div className="w-full flex justify-center mt-8">
-        <SearchBox />
-      </div>
```
Comment (farzadab) [inline packages/webapp/src/components/mainapp.tsx:14]: This didn't make sense in the main page.

File: packages/webapp/src/components/searchbox.tsx [hunk 5]
```diff
@@ -1,11 +1,156 @@
...
+        <Card.Body>
+          <Grid.Container gap={1} justify="flex-start" alignItems="center">
+            <Grid xs={2}>
+              {message.loading ? (
+                <Loading />
+              ) : (
+                <Avatar
+                  css={{ position: 'absolute', top: '0.8rem' }}
+                  size="lg"
+                  src={message.type === 'response' ? '/trefoil.png' : '/user.png'}
```
Comment (farzadab) [inline packages/webapp/src/components/searchbox.tsx]: Hardcoding the Omnibot avatar for now.

File: packages/webapp/public/trefoil.png
Comment (mdwelsh) [inline packages/webapp/public/trefoil.png:1]: We may just want to remove the use of avatars for this demo app, these two were specifically chosen as they are relevant to the Omnibus podcast, but I think dealing with the general case is outside the scope of this project at the moment :-)

File: packages/webapp/src/app/layout.tsx (hunk 1 above)
Comment (mdwelsh) [reply packages/webapp/src/app/layout.tsx:16]: You can also just put those things directly into the `RootLayout` component here (I am not sure you need the extra layer of indirection).

File: packages/webapp/src/components/searchbox.tsx [hunk 6]
```diff
@@ -1,11 +1,156 @@
-export function SearchBox() {
+'use client';
+import * as AI from 'ai-jsx/react';
+import { useState, ReactNode, useEffect } from 'react';
+
+import { Avatar, Card, Container, Input, Loading, Grid, Text } from '@nextui-org/react';
+import { BiSearch } from 'react-icons/bi';
+// import ReactMarkdown from "react-markdown";
```
Comment (mdwelsh) [inline packages/webapp/src/components/searchbox.tsx]: Remove this line?

**Review by mdwelsh — APPROVED** [review]
Nice!

File: packages/webapp/src/app/layout.tsx (hunk 1 above)
Comment (farzadab) [reply packages/webapp/src/app/layout.tsx:16]: It kept complaining. I had to add `use react` at the top and when I did that it was complaining about having an `export` in a file that has `use react`.

File: packages/webapp/public/trefoil.png
Comment (farzadab) [reply packages/webapp/public/trefoil.png:1]: Okay. Changing them to simple icons instead:
image

**Comment by farzadab** [comment]
Can I merge this?

**Comment by mdwelsh** [comment]
Please do!!

**Pull request merged**